internal/mesh/internal/mappers/sidecarproxymapper/service_endpoints_mapper.go

```diff
@@ -68,13 +58,31 @@ func (m *Mapper) MapServiceEndpointsToProxyStateTemplate(_ context.Context, _ co
 				continue
 			}
 			serviceRef := resource.Reference(serviceID, "")
-			if destination, ok := m.cache.ReadDestination(serviceRef, portName); ok {
+			if destination, ok := m.destinationsCache.ReadDestination(serviceRef, portName); ok {
 				for refKey := range destination.SourceProxies {
 					result = append(result, controller.Request{ID: refKey.ToID()})
 				}
 			}
 		}
 	}
 
+	// todo (ishustava): this is a stub for now until we implement implicit destinations.
+	// For tproxy, we generate requests for all proxy states in the cluster.
+	// This will generate duplicate events for proxies already added above,
+	// however, we expect that the controller runtime will de-dup for us.
+	rsp, err := rt.Client.List(ctx, &pbresource.ListRequest{
+		Type: types.ProxyStateTemplateType,
+		Tenancy: &pbresource.Tenancy{
+			Namespace: storage.Wildcard,
+			Partition: res.Id.Tenancy.Partition,
```

Comment:
This should be aligned, you're right. Originally, I was thinking that namespace-wide proxy configs will re-use the same resource but that has changed since as we should have `NamespaceProxyConfiguration` resource instead.